app/services/publisher_wallet_getter.rb

```diff
@@ -18,19 +18,37 @@ def perform
 
     wallet_hash = JSON.parse(wallet_response.body)
 
-    rates = wallet_hash["rates"]
-    
-    currency = wallet_hash.dig("contributions", "currency") || publisher.default_currency
-
-    if publisher.channels.verified.present?
-      channel_balances_response = PublisherBalanceGetter.new(publisher: publisher).perform
-      channel_hash = parse_channel_balances_response(channel_balances_response, rates, currency)
-
-      # Replace owner balance with sum of channel balances # TODO: Remove owner balance from the /wallet response
-      owner_balance_probi = sum_channel_balances(channel_balances_response) * 1E18
-      wallet_hash["contributions"]["probi"] = owner_balance_probi if wallet_hash.dig("contributions","probi")
+    # TODO: Remove else condition when transaction table is stable
+    if should_use_transaction_table?
+      if publisher.channels.verified.present?
+        accounts = PublisherBalanceGetter.new(publisher: publisher).perform
+
+        # Override owner balance with transaction table value
+        if wallet_hash.dig("contributions", "probi")
+          owner_balance = owner_balance_bat(accounts)
+          wallet_hash["contributions"]["probi"] = owner_balance_bat(accounts).to_d * 1E18
+          wallet_hash["contributions"]["amount"] = owner_balance 
+        end
+
+        # Convert accounts into Eyeshade::Wallet format
+        channel_hash = parse_accounts(accounts, wallet_hash)
+      else
+        channel_hash = {}
+      end
     else
+      channel_responses = {}
+      publisher.channels.verified.each do |channel|
+        identifier =  channel.details.channel_identifier
+        channel_responses[identifier] = connection.get do |request|
+          request.headers["Authorization"] = api_authorization_header
+          request.url("/v2/publishers/#{URI.escape(identifier)}/balance")
+        end
+      end
+
       channel_hash = {}
+      channel_responses.each do |identifier, response|
+        channel_hash[identifier] = JSON.parse(response.body)
+      end
```

Comment:
```
    else
      channel_responses = {}
      publisher.channels.verified.each do |channel|
        identifier =  channel.details.channel_identifier
        channel_responses[identifier] = connection.get do |request|
          request.headers["Authorization"] = api_authorization_header
          request.url("/v2/publishers/#{URI.escape(identifier)}/balance")
        end
      end

      channel_hash = {}
      channel_responses.each do |identifier, response|
        channel_hash[identifier] = JSON.parse(response.body)
      end
    end
```
Everything in this block was copy and pasted from the `PublisherWalletGetter` prior to any transaction table changes.